File: solutions/cpp/Problem26.cpp
#pragma once
/*
26. Remove Duplicates from Sorted Array

Given a sorted array nums, remove the duplicates in-place such that each element appear only once and return the new length.

Do not allocate extra space for another array, you must do this by modifying the input array in-place with O(1) extra memory.
*/
#include <iostream>
#include <vector>


class Solution {
public:
	int removeDuplicates(std::vector<int>& nums) {
		if (nums.size() == 1)
		{
			return 1;
		}

		int last;
		for (std::vector<int>::iterator it = nums.begin(); it != nums.end();)
		{
			if (it == nums.begin())
			{
				last = *it;
				it++;
				continue;
			}
			int thisVal = *it;
			if (thisVal == last)
			{
				it = nums.erase(it);
			}
			else
			{
				it++;
			}
			last = thisVal;
		}
		return nums.size();
	}
	void test()	{
		std::vector<int> vecTest;
		vecTest.push_back(1);
		vecTest.push_back(1);
		vecTest.push_back(2);
		int len = removeDuplicates(vecTest);
		for (int i = 0; i < len; i++)
		{

			std::cout << vecTest[i] << std::endl;
		}
	}
};

File: solutions/python3/problem26.py
# -*- coding: utf-8 -*-
"""
26. Remove Duplicates from Sorted Array

Given a sorted array nums,
remove the duplicates in-place such that each element appear only once and return the new length.

Do not allocate extra space for another array,
you must do this by modifying the input array in-place with O(1) extra memory.
"""


class Solution:
    def removeDuplicates(self, nums):
        if not nums:
            return 0

        length = len(nums)
        ind = 1
        last_val = nums[0]

        while ind < length:
            if nums[ind] == last_val:
                nums.pop(ind)
                length -= 1
            else:
                last_val = nums[ind]
                ind += 1

        return length
